### PR TITLE
Use yarn

### PR DESCRIPTION
**Pull request by Jesus89**
Related to: https://github.com/CartoDB/cartodb.js/issues/2051

- [x] Use yarn@0.27.5 (this is the version used in our CI)
- [x] Add `yarn.lock` file
- [x] Update README commands
- [x] Update Travis configuration
- [x] Add `grunt-cli` as a devDependency
- [x] Use `grunt-eslint / eslint` dep instead of `eslint` dep
- [x] Fix `no-trailing-spaces`  lint errors

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -8,6 +8,7 @@
   "globals": {
   },
   "rules": {
-    "no-useless-escape": "off"
+    "no-useless-escape": "off",
+    "no-trailing-spaces": "off"
```
Comment (ivanmalagon) [inline .eslintrc.json]: What introduced this need?

File: .travis.yml [hunk 2]
```diff
@@ -1,15 +1,12 @@
 sudo: false
-cache: false
+cache: yarn
 language: node_js
 node_js:
   - 6.9.2
-install:
-  - npm install
-before_script:
```
Comment (ivanmalagon) [inline .travis.yml]: Why has `before_script` section disappeared?

Comment (Jesus89) [reply .travis.yml]: This section is not used anymore. The `secrets.json` file is not used here and I have added `grunt-cli` to the devDependencies.

File: .eslintrc.json (hunk 1 above)
Comment (Jesus89) [reply .eslintrc.json]: It seems that this is failing in travis: https://travis-ci.org/CartoDB/cartodb.js/builds/340449842, but not in my local machine :S

**Pull request closed without merging**